src/metabase/models/dashboard.clj

```diff
@@ -235,6 +247,26 @@
 
 (comment moderation/keep-me)
 
+(defn archive-or-unarchive-internal-dashboard-questions!
```

Comment:
Is it possible for an archived card to get added directly to a dashboard as a dashboard question?

Reply:
Thank you for noticing this - yes, this was an oversight on my part. There are three attributes of a card that all need to be synced:

- if a DQ is archived/unarchived, we need to remove/autoplace its DashboardCards
- if a DQ's DCs are removed/added, we need to archive/unarchive it
- if a Q is turned into a DQ, we need to autoplace/unarchive it

I've done this and added a test for this case.